lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp

```diff
@@ -616,4 +624,133 @@ int amf_smf_handle_ip_address_response(
   return rc;
 }
 
+/****************************************************************************
+ **                                                                        **
+ ** Name        :  amf_pdu_session_establishment_reject()                  **
+ **                                                                        **
+ ** Description :  Send the Downlink Transport with 5GMM Cause to gnb      **
+ **                                                                        **
+ ** Inputs      :  amf_ue_ngap_id_t : pdusession response message          **
+ **                session_id		: PDU Session Inputs               **
+ **		   pti              : Procedure transaction identity       **
+ **		   5GSM cause       : 5GSM cause                           **
+ **                                                                        **
+ **  Return     :  RETURNok, RETURNerror                                   **
+ **                                                                        **
+ ***************************************************************************/
+int amf_pdu_session_establishment_reject(
+    amf_ue_ngap_id_t ue_id, uint8_t session_id, uint8_t pti, uint8_t cause) {
+  nas5g_error_code_t rc  = M5G_AS_SUCCESS;
```

Comment:
Should we not initialize to a failure and only have it go to success via amf_app_handle_nas_dl_req?